libs/dyn/pattern_trie.go

```diff
@@ -25,18 +22,18 @@ type PatternTrie struct {
 // is not supported by the [PatternTrie.SearchPath] method. We don't perform validation for this
 // case because it's not expected to arise in practice where a field is either a map or an array,
 // but not both.
-type trieNode struct {
+type TrieNode struct {
 	// If set this indicates the trie node is an anyKey node.
 	// Maps to the [AnyKey] component.
-	anyKey *trieNode
```

Comment:
Can we export AnyKey, AnyIndex, PathKey? They have well defined semantics and there could be operations not provided by methods (e.g. walk + match that I mentioned).